lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -198,65 +445,50 @@ class _Listener {
     required this.target,
     required this.handler,
     required this.useCapture,
-    required this.isNative,
   });
 
-  /// Registers a listener for the given [event] on [target] using the Dart-to-JS API.
+  /// Registers a listener for the given `event` on a `target`.
+  ///
+  /// If `passive` is null uses the default behavior determined by the event
+  /// type. If `passive` is true, marks the handler as non-blocking for the
+  /// built-in browser behavior. This means the browser will not wait for the
+  /// handler to finish execution before performing the default action
+  /// associated with this event. If `passive` is false, the browser will wait
+  /// for the handler to finish execution before performing the respective
+  /// action.
   factory _Listener.register({
     required String event,
     required DomEventTarget target,
     required DartDomEventListener handler,
     bool capture = false,
+    bool? passive,
   }) {
     final DomEventListener jsHandler = createDomEventListener(handler);
-    final _Listener listener = _Listener._(
-      event: event,
-      target: target,
-      handler: jsHandler,
-      useCapture: capture,
-      isNative: false,
-    );
-    target.addEventListener(event, jsHandler, capture);
-    return listener;
-  }
+    if (passive == null) {
+      target.addEventListener(event, jsHandler, capture);
+    } else {
+      final Map<String, Object> eventOptions = <String, Object>{
+        'capture': capture,
+        'passive': passive,
+      };
+      target.addEventListenerWithOptions(event, jsHandler, eventOptions);
+    }
 
-  /// Registers a listener for the given [event] on [target] using the native JS API.
-  factory _Listener.registerNative({
-    required String event,
-    required DomEventTarget target,
-    required DomEventListener jsHandler,
-    bool capture = false,
-    bool passive = false,
-  }) {
-    final Map<String, Object> eventOptions = <String, Object>{
-      'capture': capture,
-      'passive': passive,
-    };
-    final _Listener listener = _Listener._(
+    return _Listener._(
       event: event,
       target: target,
       handler: jsHandler,
       useCapture: capture,
-      isNative: true,
     );
-    target.addEventListenerWithOptions(event, jsHandler, eventOptions);
-    return listener;
   }
 
   final String event;
-
   final DomEventTarget target;
   final DomEventListener handler;
-
   final bool useCapture;
-  final bool isNative;
 
   void unregister() {
-    if (isNative) {
-      target.removeEventListener(event, handler, useCapture);
-    } else {
-      target.removeEventListener(event, handler, useCapture);
-    }
+    target.removeEventListener(event, handler, useCapture);
```

Comment:
There's a cool property coming: `signal` when registering the event listener that would allow us to simplify this Listener class quite a bit. Instead of having to keep a reference to the passed-in handler and useCapture values, we can just keep an instance of an "AbortSignal" object, and call its `.abort()` method when it's time to stop remove the listener. (See [this example](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#add_an_abortable_listener))

(It's Safari 15 only so maybe in a couple of years... :) )

Reply:
Interesting. Yeah, that's a much more reasonable pattern.